src/salary.cairo

```diff
@@ -0,0 +1,336 @@
+        _last_update_month_id: u32, // @dev to avoid unnecessary calculation of cum_salary
+        _master: ContractAddress, // @dev master contract address
+    }
+
+    #[event]
+    #[derive(Drop, starknet::Event)]
+    enum Event {
+        SalaryPoolUpdated: SalaryPoolUpdated,
+        SalaryClaimed: SalaryClaimed,
+    }
+
+    // @notice An event emitted whenever funds are added to salary pool.
+    #[derive(Drop, starknet::Event)]
+    struct SalaryPoolUpdated {
+        month_id: u32, 
+        guild: felt252,
+        pool_amount: u256
+    }
+
+    // @notice An event emitted whenever contribution claims salary
+    #[derive(Drop, starknet::Event)]
+    struct SalaryClaimed {
+        amount: u256,
+        recipient: ContractAddress
+    }
+    //
+    // Constructor
+    //
+
+    // @notice Contract constructor
+    #[constructor]
+    fn constructor(ref self: ContractState, owner_: ContractAddress, token_: ContractAddress, master_: ContractAddress) {
+        self._token.write(token_); // USDC
+        // self._token.write(contract_address_const::<0x053c91253bc9682c04929ca02ed00b3e423f6710d2ee7e0d5ebb06f3ecf368a8>()); // USDC
+        self._master.write(master_);
+        let mut ownable_self = Ownable::unsafe_new_contract_state();
+        ownable_self._transfer_ownership(new_owner: owner_);
+
+    }
+
+    #[external(v0)]
+    impl SalaryDistributor of super::ISalaryDistributor<ContractState> {
+        //
+        // Getters
+        //
+        fn token(self: @ContractState) -> ContractAddress {
+            self._token.read()
+        }
+
+        fn master(self: @ContractState) -> ContractAddress {
+            self._master.read()
+        }
+
+        fn get_cum_salary(self: @ContractState, contributor: ContractAddress) -> u256 {
+            InternalImpl::_calculate_cum_salary(self, contributor)
+        }
+
+        fn get_claimed_salary(self: @ContractState, contributor: ContractAddress) -> u256 {
+            self._salary.read(contributor).claimed_salary
+        }
+        // for debugging will remove after review
+        // fn get_cum_salarys(self: @ContractState, contributor: ContractAddress){
+        //     InternalImpl::get_guild_cum_salarys(self, contributor, 'dev');
+        //     InternalImpl::get_guild_cum_salarys(self, contributor, 'design');
+        //     InternalImpl::get_guild_cum_salarys(self, contributor, 'problem_solving');
+        //     InternalImpl::get_guild_cum_salarys(self, contributor, 'marcom');
+        //     InternalImpl::get_guild_cum_salarys(self, contributor, 'research');
+        // }
+
+        //
+        // Setters
+        //
+
+        fn add_fund_to_salary_pools(ref self: ContractState, month_id: u32, amounts: Array<u256>, guilds: Array<felt252>) {
+            self._only_owner();
+            let caller = get_caller_address();
+            let contract_address = get_contract_address();
```

Comment:
naming as `recipient_contract_address` will give the perception that tokens are going to some other wallet/contract instead of the same contract. will think of a better name.